docs/DeveloperGuide.md

```diff
@@ -2,21 +2,133 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links 
+to the original source as well}
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The following is our class diagram for our whole project.
 
+![Local Image](UML/CLiRental.png)
+
+### Sequence diagrams
+
+1. `Adding a customer`
+
+The following sequence diagram will explain the sequence of events after the user inputs an add-user command.
+
+![Local Image](UML/AddCustomerSequence.png)
```

Comment:
![image](https://github.com/user-attachments/assets/6c9c496e-e4b8-4dd1-a8e3-607f95d309bd)

Perhaps line 14 should be dotted instead? Given that it shows the function CustomerException returning to the activation bar of CustomerParser.

![image](https://github.com/user-attachments/assets/dc50273f-7f3a-4703-b8a5-344be00ca918)

Seems like the arrow for line 21 is in the wrong direction.